pytorch_lightning/trainer/distrib_data_parallel.py

```diff
@@ -467,7 +468,12 @@ def spawn_ddp_children(self, model):
             env_copy['LOCAL_RANK'] = f'{local_rank}'
 
             # start process
-            proc = subprocess.Popen(command, env=env_copy)
+            # if hydra is available and initialized, make sure to set the cwd correctly
+            cwd: Optional[str] = None
```

Comment:
this will break for non hydra users... no? 
shouldn't cwd be set to something for non hydra people?

Reply:
`None` is the default value for `cwd` (https://docs.python.org/3/library/subprocess.html#subprocess.Popen). If `None`, it just uses whatever the current value for cwd is.

> If cwd is not None, the function changes the working directory to cwd before executing the child. cwd can be a string, bytes or path-like object. In particular, the function looks for executable (or for the first item in args) relative to cwd if the executable path is a relative path.

Reply:
So currently, we're setting `cwd` to `None` for everyone, hence the problem for Hydra scripts.

Reply:
can we pull the cwd correctly then and also use it for non hydra users?

Reply:
sure, but what should we set it to for non hydra users? do you think we need to save the original cwd and then use that? I feel like that's going down the slippery slope b/c we can't possibly anticipate how a script will change the cwd... for hydra scripts we can be sure b/c of the way it operates...